consensus/src/dag/dag_fetcher.rs

```diff
@@ -170,17 +170,25 @@ impl DagFetcherService {
         )
     }
 
-    pub async fn start(mut self) {
-        while let Some(local_request) = self.request_rx.recv().await {
-            match self
-                .fetch(
-                    local_request.node(),
-                    local_request.responders(&self.ordered_authors),
-                )
-                .await
-            {
-                Ok(_) => local_request.notify(),
-                Err(err) => error!("unable to complete fetch successfully: {}", err),
+    pub(crate) async fn start(mut self, mut shutdown: Shutdown) {
+        while !shutdown.is_shutdown() {
+            tokio::select! {
+                Some(local_request) = self.request_rx.recv() => {
+                    match self
+                        .fetch(
+                            local_request.node(),
+                            local_request.responders(&self.ordered_authors),
+                            &mut shutdown,
```

Comment:
the nested shutdown is quite ugly, I increasingly prefer to not take the DAG into fetcher but return the nodes out. we need to duplicate the checking logic (hopefully re-use code) but feel cleaner.